starknet-e2e-test/src/starknet_sovereign.rs

```diff
@@ -0,0 +1,147 @@
+use std::fs::File;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use ethers::providers::Middleware;
+use ethers::types::{Address, I256, U256};
+use ethers::utils::keccak256;
+use mc_settlement::ethereum::client::EthereumConfig;
+use mc_settlement::ethereum::convert_felt_to_u256;
+use mp_felt::Felt252Wrapper;
+use mp_messages::{MessageL1ToL2, MessageL2ToL1};
+use mp_snos_output::SnosCodec;
+use starknet_api::hash::StarkFelt;
+use starknet_api::serde_utils::hex_str_from_bytes;
+use starknet_core_contract_client::clients::StarknetSovereignContractClient;
+use starknet_core_contract_client::interfaces::{
+    CoreContractInitData, CoreContractState, OperatorTrait, ProxyInitializeData, ProxySupportTrait,
+    StarknetMessagingTrait,
+};
+use starknet_core_contract_client::{LocalWalletSignerMiddleware, StarknetCoreContractClient};
+use starknet_ff::FieldElement;
+use zaun_sandbox::unsafe_proxy::deploy_starknet_sovereign_behind_unsafe_proxy;
+use zaun_sandbox::EthereumSandbox;
+
+pub struct StarknetSovereign {
+    _sandbox: EthereumSandbox,
+    client: StarknetSovereignContractClient,
+}
+
+impl StarknetSovereign {
+    pub fn address(&self) -> Address {
+        self.client.address()
+    }
+
+    pub fn client(&self) -> Arc<LocalWalletSignerMiddleware> {
+        self.client.client()
+    }
+
+    pub async fn deploy() -> Self {
+        // Try to attach to an already running sandbox (GitHub CI case)
+        // otherwise spawn new sandbox instance
+        let sandbox = if let Ok(endpoint) = std::env::var("ANVIL_ENDPOINT") {
+            EthereumSandbox::attach(Some(endpoint)).expect("Failed to attach to sandbox")
+        } else {
+            EthereumSandbox::spawn(None)
+        };
+
+        let client = deploy_starknet_sovereign_behind_unsafe_proxy(sandbox.client())
+            .await
+            .expect("Failed to deploy starknet contract");
+
+        Self { _sandbox: sandbox, client }
+    }
+
+    pub async fn create_settlement_conf(&self, data_path: PathBuf) -> PathBuf {
```

Comment:
A lil bit of docstring here.
It write a settlement_conf to the disk? Based on some other file it read?